CMakeLists.txt

```diff
@@ -498,543 +430,40 @@ hpx_option(
- note: enabling this also enables CUDA/HIP futures via HPX_WITH_ASYNC_CUDA"
-)
-set(HIP_OPTION_STRING "Enable compilation with HIPCC (default: OFF)")
-hpx_option(HPX_WITH_CUDA BOOL "${CUDA_OPTION_STRING}" OFF ADVANCED)
-
-# Those two will be renamed replacing CUDA with GPU
-hpx_option(HPX_WITH_COMPUTE_CUDA BOOL "${CUDA_COMPUTE_STRING}" ON ADVANCED)
-hpx_option(HPX_WITH_ASYNC_CUDA BOOL "${ASYNC_CUDA_STRING}" ON ADVANCED)
-
-# HPX_WITH_COMPUTE_CUDA requires HPX_WITH_ASYNC_CUDA
-if(HPX_WITH_COMPUTE_CUDA)
-  set(HPX_WITH_ASYNC_CUDA
-      ON
-      CACHE BOOL "${HPX_WITH_ASYNC_CUDA}" FORCE
-  )
-endif()
-
-# No need for the user to specify the option explicitly
-hpx_option(HPX_WITH_HIP BOOL "${HIP_OPTION_STRING}" OFF ADVANCED)
-if("${CMAKE_CXX_COMPILER}" MATCHES "hipcc$")
-  set(HPX_WITH_HIP
-      ON
-      CACHE BOOL "${HIP_OPTION_STRING}" FORCE
-  )
-endif()
-
-if(HPX_WITH_CUDA AND HPX_WITH_HIP)
-  hpx_error(
-    "HPX_WITH_CUDA=ON and HPX_WITH_HIP=ON. Only one of them can be on at the same time.\
- Note: HPX_WITH_HIP is automatically enabled when compiling with hipcc."
-  )
-endif()
-
 # ##############################################################################
 # pkgconfig file generation
 # ##############################################################################
-set(HPX_WITH_PKGCONFIG_DEFAULT ON)
-if(APPLE
-   OR MSVC
-   OR HPX_WITH_CUDA
-   OR HPX_WITH_HIP
-)
-  set(HPX_WITH_PKGCONFIG_DEFAULT OFF)
-endif()
-
 hpx_option(
   HPX_WITH_PKGCONFIG
   BOOL
   "Enable generation of pkgconfig files (default: ON on Linux without CUDA/HIP, otherwise OFF)"
-  ${HPX_WITH_PKGCONFIG_DEFAULT}
-  ADVANCED
-)
-
-# ##############################################################################
-# HPX datapar configuration
-# ##############################################################################
-hpx_option(
-  HPX_WITH_DATAPAR_VC
-  BOOL
-  "Enable data parallel algorithm support using the external Vc library (default: OFF)"
-  OFF
-  ADVANCED
-)
-if(HPX_WITH_DATAPAR_VC)
-  hpx_option(
-    HPX_WITH_DATAPAR_VC_NO_LIBRARY BOOL
-    "Don't link with the Vc static library (default: OFF)" OFF ADVANCED
-  )
-endif()
-
-if(HPX_WITH_DATAPAR_VC)
-  hpx_warn(
-    "Vc support is deprecated. This option will be removed in a future release. It will be replaced with SIMD support from the C++ standard library"
-  )
-  include(HPX_SetupVc)
-endif()
-if(NOT HPX_WITH_DATAPAR_VC)
-  hpx_info("No vectorization library configured")
-else()
-  hpx_option(
-    HPX_WITH_DATAPAR BOOL
-    "Enable data parallel algorithm support (default: ON)" ON ADVANCED
-  )
-endif()
-
-# ##############################################################################
-# Threadlevel Nice option
-# ##############################################################################
-hpx_option(
-  HPX_WITH_NICE_THREADLEVEL
-  BOOL
-  "Set HPX worker threads to have high NICE level (may impact performance) (default: OFF)"
-  OFF
+  ON
   ADVANCED
 )
-if(HPX_WITH_NICE_THREADLEVEL)
```

Comment:
Should we keep it for `HPXLocal_WITH_NICE_THREADLEVEL`?

Reply:
No, the definition is added in HPXLocal.